mankala/Base.ts

```diff
@@ -1,21 +1,22 @@
 ///<reference path="Driver.ts"/>
 
-module Base {
-    export interface IList {
+namespace Base {
+    export interface IList<T> {
         isHead: boolean;
-        next: IList;
-        prev: IList;
-        insertAfter(entry: IList): IList;
-        insertBefore(entry: IList): IList;
+        next: IList<T>;
+        prev: IList<T>;
+        data: T;
+        insertAfter(entry: IList<T>): IList<T>;
+        insertBefore(entry: IList<T>): IList<T>;
         item();
```

Comment:
`item()` returns `any`. I think you may want to remove `data` and update `item`.